docs/code-usage.rst

```diff
@@ -0,0 +1,178 @@
+.. index::
+   single: Phinx via Code
+
+Calling Phinx via Code
+===================================
+
+Phinx is primarily a command line tool, however Phinx does support code-based calls.
+
+.. warning::
+
+    Not all terminal commands are available to be used within code yet.
+
+Initial Setup
+--------------
+
+.. note::
+
+    For Phinx via Code to work at all, you need to ensure that Phinx is accessible via autoloading,
+    the best way to manage this would be to ``require "vendor/autoload.php";`` the composer autoloader.
+
+Phinx will require instantiating, which can be achieved by the following.
+
+.. code-block:: php
+
+    $app = new \Phinx\Console\PhinxApplication($version);
+    $phinx = new \Phinx\Wrapper\TextWrapper($app);
+
+Where ``$version`` is the version number of Phinx (eg. '0.4.1').

```

Comment:
I just saw your edit.

I agree that having an embedded one would be nice actually. Though, to me it would still be nicer if we had a Facade style class which was simply called Phinx which wrapped the TextWrapper and some of its methods

Having Phinx->migrate() is clearer than TextWrapper->getMigrate()